lib/mongoid/document.rb

```diff
@@ -98,9 +98,11 @@ def identity
     #   Person.new(:title => "Sir")
     #
     # @param [ Hash ] attrs The attributes to set up the document with.
+    # @param [ true | false ] defer_callbacks Flag specifies whether callbacks
+    #   should be run.
     #
     # @return [ Document ] A new document.
-    def initialize(attrs = nil)
+    def initialize(attrs = nil, defer_callbacks = false)
```

Comment:
In my opinion this is not ideal, I think a better way would be to use `allocate` and add a parallel method to construct instances internally.

Reply:
Oh that is a great idea

Reply:
done

Reply:
The method signature was changed a little so that I could explicitly pass the block, but it is the same for all intents and purposes.